lib/motors/controller.cpp

```diff
@@ -2,8 +2,134 @@
 
 namespace hyped::motors {
 
-Controller::Controller()
+Controller::Controller(core::ILogger &logger) : logger_(logger)
 {
 }
 
-}  // namespace hyped::motors
+void Controller::processErrorMessage(const std::uint16_t error_code)
+{
+  switch (error_code) {
+    case 0xFF01:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_CURRENT_A: Current phase A hall sensor missing or damaged");
+      break;
+    case 0xFF02:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_CURRENT_B: Current phase A hall sensor missing or damaged");
+      break;
+    case 0xFF03:
+      logger_.log(core::LogLevel::kFatal, "ERROR_HS_FET: High side Fet short circuit");
+      break;
+    case 0xFF04:
+      logger_.log(core::LogLevel::kFatal, "ERROR_LS_FET: Low side Fet short circuit");
+      break;
+    case 0xFF05:
+      logger_.log(core::LogLevel::kFatal, "ERROR_DRV_LS_L1: Low side Fet phase 1 short circuit");
+      break;
+    case 0xFF06:
+      logger_.log(core::LogLevel::kFatal, "ERROR_DRV_LS_L2: Low side Fet phase 2 short circuit");
+      break;
+    case 0xFF07:
+      logger_.log(core::LogLevel::kFatal, "ERROR_DRV_LS_L3: Low side Fet phase 3 short circuit");
+      break;
+    case 0xFF08:
+      logger_.log(core::LogLevel::kFatal, "ERROR_DRV_HS_L1: High side Fet phase 1 short circuit");
+      break;
+    case 0xFF09:
+      logger_.log(core::LogLevel::kFatal, "ERROR_DRV_HS_L2: High side Fet phase 2 short circuit");
+      break;
+    case 0xFF0A:
+      logger_.log(core::LogLevel::kFatal, "ERROR_DRV_HS_L3: High side Fet phase 3 short circuit ");
+      break;
+    case 0xFF0B:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_MOTOR_FEEDBACK: Wrong feedback selected (check feedback type)");
+      break;
+    case 0xFF0C:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_DC_LINK_UNDERVOLTAGE: DC voltage not applied to bridge or too low");
+      break;
+    case 0xFF0D:
+      logger_.log(core::LogLevel::kFatal, "ERROR_PULS_MODE_FINISHED: Pulse mode finished");
+      break;
+    case 0xFF0E:
+      logger_.log(core::LogLevel::kFatal, "ERROR_APP_ERROR");
+      break;
+    case 0xFF0F:
+      logger_.log(core::LogLevel::kFatal, "ERROR_EMERGENCY_BUTTON: Emergency button pressed");
+      break;
+    case 0xFF10:
+      logger_.log(core::LogLevel::kFatal,
+                  "ERROR_CONTROLLER_OVERTEMPERATURE: Controller overtemperature");
+      break;
+    case 0x3210:
```

Comment:
Is this correct? Looks very different to all the other values.

Reply:
<img width="657" alt="image" src="https://user-images.githubusercontent.com/50985559/208084464-677869ba-05b3-4726-abfd-4391fc92d1b4.png">
yep!